contracts/MiniMeToken.sol

```diff
@@ -544,6 +550,32 @@ contract MiniMeToken is Controlled {
         transfersEnabled = _transfersEnabled;
     }
 
+
+////////////////
+// Arbitrum bridge functions
+////////////////
+
+    modifier onlyBridge {
+        require(msg.sender == bridge);
+        _;
+    }
+
+    /// @notice Changes the bridge with ability to assign and lock tokens
+    /// @param _bridge The new controller of the contract
+    function changeBridge(address _bridge) onlyController  public {
```

Comment:
I would like to know your thinking process to only allow set up the bridge address after contract creation. Did you consider configure it on the constructor?

If I understand correctly. If we follow this pattern: https://github.com/OffchainLabs/arbitrum/blob/master/packages/arb-bridge-peripherals/contracts/tokenbridge/arbitrum/StandardArbERC20.sol#L45. We can create our own L2Gateway that allows users to migrate their `L1ERC20` to an `L2Minime` Governance token. Where our Gateway implementation call `createCloneToken` making sure we check the bridge was initialized like in: https://github.com/OffchainLabs/arbitrum/blob/master/packages/arb-bridge-peripherals/contracts/tokenbridge/libraries/L2GatewayToken.sol#L54

Maybe I am missing the point of how Gateways should be used.

Reply:
Yah this is possible, and a good idea if others want to emulate our issuance setup. However, someone will need to be set as the token controller upon creation and they will need to manually generate tokens to sync up with the L1 ERC20 token (if they use the same issuance process). I'm not currently sure how to handle this. We could create a separate contract that the 1Hive DAO has permissions on to set the token controller initially and generate tokens, but it's more involved, might be better to see if there's interest before developing something like that.

Reply:
Good point. I did not have in mind the token controller. Indeed it seems a much more involved process. I agree that is better to keep it as simple as possible for now.